tests/test_parsing.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+from examples.eval_model import setup_args
+import parlai.utils.testing as testing_utils
+import unittest
+
+
+class TestRecommended(unittest.TestCase):
+    """Basic tests on the recommended values of parser."""
+
+    def test_help(self):
+        """Test help message of parser"""
+        parser = setup_args()
+        parser.set_defaults(
+            task='integration_tests',
+            model='transformer/polyencoder',
```

Comment:
So this thing has a lot of dependencies, and we really only need to test one small behavior. So rather than an integration test, we'd prefer a smaller unit test.

So probably better would be to just:
- create a parlai parser
- add an argument with a recommended value
- add an argument group, and in that add an argument with a recommended value
- then do the format_help stuff like what you have below. The string code you've written is a LITTLE messy, maybe we can find to have a better matcher? A regex might be the way to go. With some flag, you can have them support newlines. something like `r'--my-option[^\n]*\n[^\n]*(recommended: myrecommendation)'`

I'm also pretty sure we have a `test_params.py` which is where this test can really go

Reply:
buried deep in ArgumentParser there might be a way to get the help message for just one option too, which would also work.

Reply:
Tried looking for it could't find one, the current check(one I just pushed) seems concise and neat, what do you think